skill-tree/src/main/java/com/RDS/skilltree/Health/HealthCheckController.java

```diff
@@ -0,0 +1,29 @@
+package com.RDS.skilltree.Health;
+
+import com.RDS.skilltree.metrics.MetricService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Map;
+
+@RestController
+@RequestMapping("/v1/health")
+@Slf4j
+public class HealthCheckController {
+
+    private final MetricService metricService;
+
+    @Autowired
+    public HealthCheckController(MetricService metricService) {
+        this.metricService = metricService;
+    }
+
+    @GetMapping("")
```

Comment:
Just a thought, if we are using actuator, why not use the actuator `/health` endpoint? Do we need a separate health controller in that case?

Reply:
Hi @debanjanc01 , 

Thank you for your review and the insightful questions.

Actuator's /health endpoint is comprehensive for general health status, our custom implementation lets us tailor the health information and format it according to our specific requirements.

Spring Boot Actuator's /health endpoint can provide uptime information, but this is not included by default. The /health endpoint typically shows basic health information about the application, like database connectivity, disk space, and other custom health indicators. 

`/actuator/health`

> {"status":"UP"}

`/actuator/health/db`

> {"status":"UP","details":{"database":"MySQL","validationQuery":"isValid()"}}

`/actuator/health/diskSpace`

> {"status":"UP","details":{"total":499963174912,"free":99603329024,"threshold":10485760,"path":"/~/rds-skill-tree-backend/.","exists":true}}


On the other hand, the /metrics endpoint `/actuator/metrics/process.uptime` will give this result by default:

> {"name":"process.uptime","description":"The uptime of the Java virtual machine","baseUnit":"seconds","measurements":[{"statistic":"VALUE","value":22.649}],"availableTags":[]}

Hence, our custom approach will allow for more specific control over the health data being exposed, which includes the uptime. It will also help in formatting the output, and, in future, include additional details specific to the application's needs that are not provided by Actuator's standard endpoints.

Reply:
Hi @iRohitGaur, in that case I'm assuming that the actuator has been added for current/future functionalities, while keeping the custom health controller.

With actuator in place, since the `process.uptime` metric gives the value for uptime, I'm trying to understand the reason for registering a `custom.uptime` gauge. Are we looking for a different value?

Reply:
Hi @debanjanc01 ,

Thank you for pointing this out. I have updated the code and removed the unnecessary `custom.uptime` guage.